### DISCUSSION TITLE
integrating/preserved-files

### DISCUSSION THREAD

**Discussion by giscus** [velopack/velopack.docs#25] [General]
# integrating/preserved-files

Your application binary files get completely replaced during updates, so you should not store important files in the same directory as your application binary files.

https://docs.velopack.io/integrating/preserved-files

**Comment by cesar-albusac** [comment]
Hello, regarding the application settings:

`If you want to create settings that persist through updates, but are erased when the app is uninstalled, on Windows you should store them one level up (..\) outside of the current dir.`

I am trying to keep a config file outsude the "current" dir: 

%LocalAppData%\MyApplication\ will contain:
- current folder
- packages folder
- Update.exe
- config (File I place there I want to be kept). 

However, everytime I upgrade/uninstall the application, everything is removed.

**Reply by caesay** [reply]
If your files are in a structure like you supplied, it will not be removed upon update. It _will_ be removed upon install/uninstall as described on this documentation page. If you need something that survives install/uninstall, you should store your config in `%appdata%\MyApplication` instead of `%localappdata%\MyApplication`

**Reply by cesar-albusac** [reply]
You are right. With the auto update, it's keeping the files. 

I was testing wrongly by clicking in the newer win-Setup.exe , And I got the question : 

"An older version of X is installed. would you like to update from 1.0.3 to 1.0.4? 

By clicking "Update to version 1.0.4", the config file were gone.

**Comment by Summpot** [comment]
Should an update done by clicking Setup.exe and then clicking the "Update" option be considered as an "update"?
I noticed that the "Update" option in the installer seems to have the same behavior as the "Repair" option in that it deletes files that should be preserved in an "update" (any files other than current, packages, Update.exe), is this expected behavior?

**Reply by caesay** [reply]
Setup.exe always performs a repair/reinstall (regardless of whether the message in the dialog says Upgrade, Repair, or Downgrade). Initially Setup.exe always said "Repair" there, but by popular request it was changed. https://github.com/velopack/velopack/issues/3

**Reply by Summpot** [reply]
Thanks for your reply, should this be mentioned somewhere in the documentation?

**Comment by rezaelahi-ui** [comment]
Hello,

Thank you for creating such a great tool! I’m using Velopack to handle updates for my desktop application. My application creates a SQLite database file (database.db) the first time it runs. I want to ensure this file is not deleted or overwritten during updates.

I’ve read about the persist option in the velopack.json file, but I want to confirm if there are any additional steps or considerations to ensure this works properly.

Could you please provide guidance or best practices for handling such files during updates?

Thank you in advance!

Best regards,

**Reply by rezaelahi-ui** [reply]
I Use Portable Version

**Reply by Keboo** [reply]
A better place to ask will be our [discord](https://discord.gg/CjrCrNzd3F).

With that said, you don't want to be creating your SQLite file _within_ the same directory that contains your application. Applications should store data like that database in a different location. If you are on Windows this often means storing it within the %APPDATA% or %LOCALAPPDATA% paths. For things like MAUI applications, [there are APIs](https://learn.microsoft.com/dotnet/maui/platform-integration/storage/file-system-helpers?view=net-maui-9.0&tabs=windows&WT.mc_id=DT-MVP-5003472#platform-differences) for finding this directory.

**Comment by SebastianDevelops** [comment]
Hi, I'm not sure if this is a silly question but how does one go about doing the below:

> If you want to create settings that persist through updates, but are erased when the app is uninstalled, on Windows you should store them one level up (..\) outside of the current dir.

My application currently keeps the application files even after the user uninstalls the application, I want this to be removed when uninstalling but kept during updates. TIA!

**Reply by caesay** [reply]
Your application is installed into `%LocalAppData%\YourAppId`. Your actual compiled files exist in `%LocalAppData%\YourAppId\current`. Everything in the `current` dir is replaced on every update. Everything in `%LocalAppData%\YourAppId` is deleted during uninstall.

**Comment by AscarGb** [comment]
Is there a way to keep certain files from being deleted from `current`? For example the `wwwroot` folder in `blazor maui hybrid` should be there.

**Reply by caesay** [reply]
The `current` folder is entirely replaced each time you update, as described here. This will not change. In your blazor example, I'd imagine that your wwwroot should be present in each release (so even though the old wwwroot is gone, the wwwroot folder from the new release will now be in place.)

**Reply by AscarGb** [reply]
In `wwwroot` this is the only place available for storing files, other methods greatly reduce performance.
